pallets/rewards/src/lib.rs

```diff
@@ -0,0 +1,535 @@
+					ensure!(!user_reward_info.is_initialized, Error::<T>::RewardsAlreadyUnlocked);
+					//transfer funds from pallet account to users account
+					ensure!(
+						Self::transfer_pdex_rewards(
+							&Self::get_pallet_account(),
+							&user,
+							user_reward_info.total_reward_amount
+						)
+						.is_ok(),
+						Error::<T>::TransferFailed
+					);
+
+					//lock funds in users account
+					T::NativeCurrency::set_lock(
+						REWARDS_LOCK_ID,
+						&user,
+						user_reward_info.total_reward_amount.saturated_into(),
+						WithdrawReasons::TRANSFER,
+					);
+					user_reward_info.is_initialized = true;
+					Ok(())
+				} else {
+					//sanity check
+					Err(Error::<T>::UserNotEligible)
+				}
+			})?;
+			Self::deposit_event(Event::UserUnlockedReward { user, reward_id });
+			Ok(())
+		}
+
+		/// The user will use the extrinsic to claim rewards
+		/// # Parameters
+		/// * `origin`: The users address which has been mapped to reward id
+		/// * `id`: The reward id
+		#[pallet::weight(10_000)]
+		pub fn claim(origin: OriginFor<T>, reward_id: u32) -> DispatchResult {
+			let user: T::AccountId = ensure_signed(origin)?;
+
+			//check if given id valid or not
+			ensure!(
+				<InitializeRewards<T>>::contains_key(reward_id),
+				Error::<T>::RewardIdNotRegister
+			);
+
+			//check if user is added in reward list
+			ensure!(<Distributor<T>>::contains_key(reward_id, &user), Error::<T>::UserNotEligible);
+
+			<Distributor<T>>::mutate(reward_id, user.clone(), |user_reward_info| {
+				if let Some(reward_info) = <InitializeRewards<T>>::get(reward_id) {
+					if let Some(user_reward_info) = user_reward_info {
+						//check if user has initialize rewards or not
+						ensure!(
+							user_reward_info.is_initialized,
+							Error::<T>::UserHasNotInitializeClaimRewards
+						);
+
+						let mut rewards_claimable: u128 = 0_u128.saturated_into();
+
+						//if initial rewards are not claimed add it to claimable rewards
+						if !user_reward_info.is_initial_rewards_claimed {
+							rewards_claimable =
+								user_reward_info.initial_rewards_claimable.saturated_into::<u128>();
+						}
+
+						//calculate the number of blocks the user can claim rewards
+						let current_block_no: u128 =
+							<frame_system::Pallet<T>>::block_number().saturated_into();
+						let last_reward_claimed_block_no: u128 =
+							user_reward_info.last_block_rewards_claim.saturated_into();
+						let unclaimed_blocks: u128 =
+							min(current_block_no, reward_info.end_block.saturated_into::<u128>())
+								.saturating_sub(last_reward_claimed_block_no);
+
+						// add the unclaimed block rewards to claimable rewards
+						rewards_claimable = rewards_claimable.saturating_add(
+							user_reward_info
+								.factor
+								.saturated_into::<u128>()
+								.saturating_mul(unclaimed_blocks),
+						);
+
+						//ensure total_rewards_claimable - rewards_claimed >= rewards_claimable
+						//sanity check
+						ensure!(
```

Comment:
This condition should never trigger unless there is anything wrong in the calculation or governance upgrades the storage. I can't just put greater than, as the user will not be able to claim the last claimable reward. 
If a user tries to claim a reward when all rewards are already claimed rewards_claimable will have a value of zero and the `MIN_REWARDS_CLAIMABLE_AMOUNT` ensure will trigger.

Reply:
>= will do